src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringMethodTranslator.cs

```diff
@@ -189,6 +192,19 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
                         },
                         typeof(int)),
                     _sqlExpressionFactory.Constant(0));
+
+                if (pattern is SqlConstantExpression constantPattern)
```

Comment:
Check for a constant pattern earlier and do not allocate an `STRPOS` expression.

Reply:
Unless I'm missing something that would require checking for constant twice, or duplicating the code for constructing the STRPOS expression; and the only advantage would be to save a single expression construction during compilation when the pattern is an empty string (an edge case if there ever was one...). So I think it's better to keep the code simple as it is...